pynucastro/rates/library.py

```diff
@@ -397,31 +397,43 @@ def backward(self):
         return only_bwd
 
     def derived_forward(self):
+        """
+        We exclude the weak and tabular rates from the .foward() library.
+        """
 
         collect_rates = []
         onlyfwd = self.forward()
 
         for r in onlyfwd.get_rates():
+
             try:
                 DerivedRate(r, use_pf=True, use_A_nuc=True)
-            except AssertionError:
+            except ValueError:
                 continue
-            collect_rates.append(r)
+            else:
+                collect_rates.append(r)
 
         list1 = Library(rates=collect_rates)
         return list1
 
     def derived_backward(self, use_pf=False, use_A_nuc=False):
+        """
+        We apply the detailed balance calculations over the selected .derived_forward()
+        library rates
```

Comment:
also make this more descriptive.  It is not clear to me what this function does now